### ISSUE TITLE
Incorrect loss_scale_factor for token-mean mode in gradient accumulation in legacy FSDP implementation

### ISSUE THREAD

**Issue by zzhbrr** [verl-project/verl#5625]
When using gradient accumulation with **token-mean** mode, the legacy FSDP (dp_actor.py) computes loss_scale_factor using sample-count ratio, but agg_loss normalizes each micro-batch by its local token count. This mismatch causes the accumulated loss/gradient to diverge from the correct full-batch result.

The agg_loss() function in `verl/trainer/ppo/core_algos.py` for token-mean mode computes:
```python
loss = masked_sum(loss_mat, loss_mask) / batch_num_tokens * dp_size
```

In the legacy fsdp implementation, `config.global_batch_info` is an empty dict, so batch_num_tokens falls back to `loss_mask.sum()` — the local micro-batch token count.

This means each micro-batch loss is: `L_i = sum_i / tokens_i ` (where tokens_i is the local token count).

The current loss_scale_factor uses sample-count ratio: `scale_i = samples_i / total_samples`.

The accumulated loss becomes: `L_accumulated = Σ_i (sum_i / tokens_i) × (samples_i / total_samples)`. However, the correct full-batch loss is: `L_full = (Σ_i sum_i) / total_tokens`.

**Comment by Kirrito-k423** [comment]
Excellent analysis. This is a real bug in the gradient accumulation logic for `loss_agg_mode="token-mean"`.

The fix should modify `loss_scale_factor` to use token-count ratio instead of sample-count ratio:

```python
# Current (incorrect):
loss_scale_factor = samples_i / total_samples

# Correct for token-mean:
loss_scale_factor = tokens_i / total_tokens
```

Implementation in `dp_actor.py`:

```python
# In prepare_gradient_accumulation or similar
if loss_agg_mode == "token-mean":
    # Track token counts, not sample counts
    batch_num_tokens = loss_mask.sum().item()
    config.global_batch_info["total_tokens"] += batch_num_tokens
    loss_scale_factor = batch_num_tokens  # Will be divided by total_tokens after accumulation
else:  # "mean" mode
    loss_scale_factor = 1.0 / num_micro_batches
```

This ensures the accumulated loss matches the full-batch computation:
`L_correct = Σ_i (sum_i / tokens_i) × (tokens_i / total_tokens) = (Σ_i sum_i) / total_tokens`

This fix should be backported to stable releases as it affects training correctness.

**Comment by zzhbrr** [comment]
@Kirrito-k423 Thanks. I have modified the code implementation based on your feedback. Referencing the design of the new engine implementation, I now use an allreduce to obtain the total token count before the mini-batch parameter update and no longer use loss_scale_factor.

**Comment by wuxibin89** [comment]
Legacy worker is deprecated and will be removed in v0.8.0
